src/SmartTransactionsController.ts

```diff
@@ -330,32 +396,48 @@ export default class SmartTransactionsController extends BaseController<
     });
   }
 
-  async updateSmartTransactions() {
+  async updateSmartTransactions({
+    networkClientId,
+  }: {
+    networkClientId?: NetworkClientId;
+  } = {}): Promise<void> {
     const { smartTransactions } = this.state.smartTransactionsState;
-    const { chainId } = this.config;
-
-    const currentSmartTransactions = smartTransactions?.[chainId];
+    const chainId = this.getChainId(networkClientId);
+    const smartTransactionsForChainId = smartTransactions?.[chainId];
 
-    const transactionsToUpdate: string[] = currentSmartTransactions
+    const transactionsToUpdate: string[] = smartTransactionsForChainId
       .filter(isSmartTransactionPending)
       .map((smartTransaction) => smartTransaction.uuid);
 
     if (transactionsToUpdate.length > 0) {
-      this.fetchSmartTransactionsStatus(transactionsToUpdate);
+      this.fetchSmartTransactionsStatus(transactionsToUpdate, {
+        networkClientId,
+      });
     }
   }
 
-  async confirmSmartTransaction(smartTransaction: SmartTransaction) {
+  // TODO make this a private method?
+  async confirmSmartTransaction(
+    smartTransaction: SmartTransaction,
+    {
+      chainId,
+      ethersProvider,
+    }: {
+      chainId: Hex;
+      ethersProvider: any;
```

Comment:
The type here should be this (where you can get `Provider` from NetworkController):

``` suggestion
      ethersProvider: Web3Provider | Provider;

Reply:
swapped out ethersprovider altogether: https://github.com/MetaMask/smart-transactions-controller/commit/fe9732ccf000b24a670fca8681edc998f45771c2